plugin/pkg/scheduler/algorithm/predicates/error.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package predicates
+
+import "fmt"
+
+var (
+	ErrPodExceedsMaxPodNumber = newResourceUnfitError("PodExceedsMaxPodNumber")
+	ErrPodExceedsFreeCPU      = newResourceUnfitError("PodExceedsFreeCPU")
+	ErrPodExceedsFreeMemory   = newResourceUnfitError("PodExceedsFreeMemory")
+)
+
+// ErrResourceUnfit is an error type that indicates what kind of resource limit is
+// hit and caused the unfitting failure.
+type ErrResourceUnfit struct {
+	reason string
```

Comment:
If you expect the error to escape the package, it's actually a good idea to make the internal fields public so that users can get at them without parsing the error message.